addons/website/wizard/portal_wizard.py

```diff
@@ -0,0 +1,35 @@
+# -*- encoding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo.tools.translate import _
+from odoo import models
+
+class PortalWizardUser(models.TransientModel):
+    _inherit = ['portal.wizard.user']
+
+    def _get_similar_user_domain(self, email):
+        """ Return the domain needed to find the users that have the same email
+        than the current partner depending on its linked website
+        characteristics.
+        :param string email: the email of the current partner
+        """
+        similar_user_domain = super(PortalWizardUser, self)._get_similar_user_domain(email)
+        if self.partner_id.website_id:
+            # If the partner is linked to a website, only keep the users that
+            # are linked to this website.
+            similar_user_domain.append(('website_id.id', '=', self.partner_id.website_id.id))
```

Comment:
Indeed, It seems to work as well.  Because `website_id` of `res_users` is a field of type `fields.Many2one('website')` I did not know that we could use it as is in a `search()` :+1: